.github/workflows/flutter-action-formatting-and-unit-testing.yml

```diff
@@ -0,0 +1,36 @@
+name: Flutter actions formatting and unit testing
+
+on:
+  push:
+    branches-ignore:
+      - 'master'
+    paths-ignore:
+      - 'documentation/**'
+      - 'LICENSE'
+      - 'README.md'
+      - '.gitignore'
+      - 'pre-commit-config.yaml'
+
+defaults:
```

Comment:
* Rubs hands greedily * Moved this here so each step can have a label, which will help when trying to figure why the build failed.